front/app/containers/Admin/pagesAndMenu/containers/ContentBuilder/components/CraftComponents/OpenToParticipation/ProjectCarrousel/index.tsx

```diff
@@ -0,0 +1,254 @@
+import { MiniProjectData } from 'api/projects_mini/types';
+
+import { DEFAULT_PADDING } from 'components/admin/ContentBuilder/constants';
+
+import { useIntl } from 'utils/cl-intl';
+
+import { CARD_GAP, CARD_WIDTH } from './constants';
+import HorizontalScroll from './HorizontalScroll';
+import LightProjectCard from './LightProjectCard';
+import messages from './messages';
+
+const Container = styled(Box)`
+  .scroll-button {
+    opacity: 0;
+    transition: opacity 0.3s;
+    cursor: pointer;
+  }
+
+  &:hover {
+    .scroll-button {
+      opacity: 1;
+    }
+  }
+`;
+
+const ProjectContainer = styled.div`
+  scroll-snap-align: start;
+`;
+
+const SkipButton = styled.button`
+  position: absolute;
+  left: 0;
+  top: 0;
+  opacity: 0;
+  z-index: 2;
+  padding: 8px;
+  background: ${colors.white};
+
+  &:focus {
+    top: 40px;
+    opacity: 1;
+  }
+`;
+
+interface Props {
+  title: string;
+  projects: MiniProjectData[];
+  hasMore: boolean;
+  onLoadMore: () => void;
+}
+
+const ProjectCarrousel = ({ title, projects, hasMore, onLoadMore }: Props) => {
+  const [scrollContainerRef, setScrollContainerRef] = useState<
+    HTMLDivElement | undefined
+  >(undefined);
+  const [showPreviousButton, setShowPreviousButton] = useState(false);
+  const [showNextButton, setShowNextButton] = useState(false);
+  const isSmallerThanPhone = useBreakpoint('phone');
+  const { formatMessage } = useIntl();
+  const instanceId = useInstanceId();
+  const endId = `end-carrousel-${instanceId}`;
+
+  const { ref } = useInView({
+    onChange: (inView) => {
+      if (inView && hasMore) {
+        onLoadMore();
+      }
+    },
+  });
+
+  const handleButtonVisiblity = useCallback(
+    (ref: HTMLDivElement, hasMore: boolean) => {
+      if (isSmallerThanPhone) return;
+
+      const scrollLeft = ref.scrollLeft;
+      const scrollWidth = ref.scrollWidth;
+      const clientWidth = ref.clientWidth;
+
+      const atEnd = scrollLeft >= scrollWidth - clientWidth;
+
+      if (atEnd && !hasMore) {
+        setShowNextButton(false);
+      } else {
+        setShowNextButton(true);
+      }
+
+      const atStart = scrollLeft === 0;
+      setShowPreviousButton(!atStart);
+    },
+    [isSmallerThanPhone]
+  );
+
+  useEffect(() => {
+    if (!scrollContainerRef) return;
+
+    const handler = debounce(() => {
+      handleButtonVisiblity(scrollContainerRef, hasMore);
+    }, 100);
+
+    scrollContainerRef.addEventListener('scroll', handler);
+
+    return () => {
+      scrollContainerRef.removeEventListener('scroll', handler);
+    };
+  }, [scrollContainerRef, hasMore, handleButtonVisiblity]);
+
+  const skipCarrousel = () => {
+    const element = document.getElementById(endId);
+    element?.setAttribute('tabindex', '-1');
+    element?.focus();
+  };
+
+  const handleKeyDown = (e: React.KeyboardEvent<any>) => {
+    if (e.code === 'Escape') {
+      skipCarrousel();
+    }
+
+    if (e.code === 'Tab' && scrollContainerRef) {
+      setTimeout(() => {
```

Comment:
This has to be wrapped in a `setTimeout`, because otherwise it's trying to focus on the next element at the same time as trying to scroll, which for some reason doesn't work. But if you wait 50ms before scrolling it does.